packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/RealmListTests.kt

```diff
@@ -411,6 +414,97 @@ class RealmListTests {
         }
     }
 
+    @Test
+    fun query_objectList() = runBlocking {
+        val container = realm.write {
+            copyToRealm(
+                RealmListContainer().apply {
+                    (1..5).map {
+                        objectListField.add(RealmListContainer().apply { stringField = "$it" })
+                    }
+                }
+            )
+        }
+        val objectListField = container.objectListField
+        assertEquals(5, objectListField.size)
+
+        val all: RealmQuery<RealmListContainer> = container.objectListField.query()
+        val ids = (1..5).map { it.toString() }.toMutableSet()
+        all.find().forEach { assertTrue(ids.remove(it.stringField)) }
+        assertTrue { ids.isEmpty() }
+
+        container.objectListField.query("stringField = $0", 3.toString()).find().single()
+            .run { assertEquals("3", stringField) }
+    }
+
+    @Test
+    fun query_embeddedObjectList() = runBlocking {
+        val container = realm.write {
+            copyToRealm(
+                RealmListContainer().apply {
+                    (1..5).map {
+                        embeddedRealmObjectListField.add(EmbeddedLevel1().apply { id = it })
+                    }
+                }
+            )
+        }
+        val embeddedLevel1RealmList = container.embeddedRealmObjectListField
+        assertEquals(5, embeddedLevel1RealmList.size)
+
+        val all: RealmQuery<EmbeddedLevel1> = container.embeddedRealmObjectListField.query()
+        val ids = (1..5).toMutableSet()
+        all.find().forEach { assertTrue(ids.remove(it.id)) }
+        assertTrue { ids.isEmpty() }
+
+        container.objectListField.query("stringField = $0", 3.toString()).find().single()
+            .run { assertEquals("3", stringField) }
+    }
+    @Test
+    fun query_throwsOnSyntaxError() = runBlocking {
+        val instance = realm.write { copyToRealm(RealmListContainer()) }
+        assertFailsWithMessage<IllegalArgumentException>("syntax error") {
+            instance.objectListField.query("ASDF = $0 $0")
+        }
+        Unit
+    }
+
+    @Test
+    fun query_throwsOnUnmanagedList() = runBlocking {
+        realm.write {
+            val instance = RealmListContainer()
+            copyToRealm(instance)
+            assertFailsWithMessage<IllegalArgumentException>("Unmanaged list cannot be queried") {
+                instance.objectListField.query()
+            }
+            Unit
+        }
+    }
+
+    @Test
+    fun query_throwsOnDeletedList() = runBlocking {
+        realm.write {
+            val instance = copyToRealm(RealmListContainer())
+            val objectListField = instance.objectListField
+            delete(instance)
+            assertFailsWithMessage<IllegalStateException>("Access to invalidated Collection") {
+                objectListField.query()
+            }
+        }
+        Unit
+    }
+
+    @Test
+    fun query_throwsOnClosedList() = runBlocking {
```

Comment:
There seem to be a test missing showing that the flow complete if you delete the object containing the list while the flow is running.